### PR TITLE
Key Value

### PR DESCRIPTION
**Pull request by scottf**

### CODE REVIEW

File: src/NATS.Client/KeyValue/KeyValue.cs [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+                int timeout = 5000; // give plenty of time for the first
+                while (true)
+                {
+                    Msg m = sub.NextMessage(timeout);
+                    action.Invoke(m);
+                    timeout = 100; // the rest should come pretty quick
+                }
+            }
+            catch (NATSTimeoutException) { /* no more messages */ }
+            sub.Unsubscribe();
```
Comment (sixlettervariables) [inline src/NATS.Client/KeyValue/KeyValue.cs]: `sub.Unsubscribe` should be in a finally block.

File: src/NATS.Client/KeyValue/KeyValue.cs [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+                .Build();
+            IJetStreamPushSyncSubscription sub = js.PushSubscribeSync(subject, pso);
+            try
+            {
+                int timeout = 5000; // give plenty of time for the first
+                while (true)
+                {
+                    Msg m = sub.NextMessage(timeout);
+                    action.Invoke(m);
+                    timeout = 100; // the rest should come pretty quick
```
Comment (sixlettervariables) [inline src/NATS.Client/KeyValue/KeyValue.cs]: These timeout values are curiously fixed. Is timing out truly our only means of detecting we've hit the end?

File: src/NATS.Client/KeyValue/KeyValueWatchSubscription.cs [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using NATS.Client.Internals;
+using NATS.Client.JetStream;
+
+namespace NATS.Client.KeyValue
+{
+    public class KeyValueWatchSubscription
```
Comment (sixlettervariables) [inline src/NATS.Client/KeyValue/KeyValueWatchSubscription.cs]: Consider making this IDisposable and having Dispose call Unsubscribe.

File: src/NATS.Client/KeyValue/KeyValue.cs (hunk 1 above)
Comment (scottf) [reply src/NATS.Client/KeyValue/KeyValue.cs]: fixing

File: src/NATS.Client/KeyValue/KeyValue.cs (hunk 2 above)
Comment (scottf) [reply src/NATS.Client/KeyValue/KeyValue.cs]: Adding check for Pending == 0

File: src/NATS.Client/KeyValue/KeyValueWatchSubscription.cs (hunk 3 above)
Comment (scottf) [reply src/NATS.Client/KeyValue/KeyValueWatchSubscription.cs]: done

**Review by aricart — APPROVED** [review]
LGTM

**Pull request merged**